labonneboite/common/models/form.py

```diff
@@ -0,0 +1,116 @@
+import datetime, enum
+
+from sqlalchemy.dialects import mysql
+from sqlalchemy import Column, ForeignKey
+from sqlalchemy import Boolean, DateTime, Integer, String, Text
+from sqlalchemy.types import Enum
+
+from labonneboite.common import mapping as mapping_util
+from labonneboite.common.models import Office
+from labonneboite.common.database import Base
+from labonneboite.common.models.base import CRUDMixin
+from labonneboite.web.contact_form import forms
+from labonneboite.common.models import OfficeAdminUpdate
+
+class RecruiterMessageCommon():
+    # Identification part
+    id = Column(Integer, primary_key=True)
+    siret = Column(Text, nullable=False)
+    requested_by_first_name = Column(mysql.TINYTEXT, nullable=False)
+    requested_by_last_name = Column(mysql.TINYTEXT, nullable=False)
+    requested_by_email = Column(mysql.TINYTEXT, nullable=False)
+    requested_by_phone = Column(mysql.TINYTEXT, nullable=False)
```

Comment:
Pourquoi tous ces champs ne sont-ils pas nullable ? Du point de vue de la db, c'est mieux de stocker des NULL au lieu de stocker des strings vides.

Reply:
Ces champs sont obligatoires au niveau du formulaire, ils servent à contacter le recruteur.

Reply:
Ca n'a rien à voir. Si la colonne n'est pas nullable, on peut quand même avoir des valeurs vides dans la colonne (""). Si tu veux empêcher l'utilisateur de stocker des chaînes vides dans la db, il faut ajouter une contrainte au niveau de flask : https://stackoverflow.com/questions/20718469/validation-in-sqlalchemy#20718668

Imaginons qu'une table T grossise énormément et contienne 10M d'entrées. Un jour on rajoute une colonne `Column(mysql.TINYTEXT, nullable=False)` à cette table. Alors il faudra créer 10M d'entrées avec une valeur par défaut. Au contraire, si on définit `nullable=True` sur cette nouvelle colonne, alors l'ajout des 10M de lignes se fera rapidement puisque les valeurs seront NULL. 

Bref, les colonnes nullables, c'est bien.

Reply:
Même si la contrainte est définie au niveau du formulaire ?

Reply:
:+1: